zerver/lib/events.py

```diff
@@ -79,21 +79,23 @@ def get_custom_profile_field_values(realm_id: int) -> Dict[int, Dict[str, Any]]:
 def get_raw_user_data(realm: Realm, user_profile: UserProfile, client_gravatar: bool,
                       include_custom_profile_fields: bool=True) -> Dict[int, Dict[str, str]]:
     user_dicts = get_realm_user_dicts(realm.id)
-
+    custom_profile_field_data = None
     if include_custom_profile_fields:
         profiles_by_user_id = get_custom_profile_field_values(realm.id)
-    else:
-        profiles_by_user_id = defaultdict(dict)
-    return {
-        row['id']: format_user_row(realm,
-                                   acting_user = user_profile,
-                                   row=row,
-                                   client_gravatar= client_gravatar,
-                                   custom_profile_field_data = profiles_by_user_id.get(row['id'], {}),
-                                   include_custom_profile_fields=include_custom_profile_fields,
-                                   )
-        for row in user_dicts
-    }
+    result = {}
+    for row in user_dicts:
+        try:
+            custom_profile_field_data = profiles_by_user_id.get(row['id'], {})
+        except NameError:
+            pass
```

Comment:
This part doesn't really seem nice. Can we use something like:
```python3
    if include_custom_profile_fields:
        profiles_by_user_id = get_custom_profile_field_values(realm.id)
    else:
        profiles_by_user_id = defaultdict(None)
```
To avoid this try/except block in the first place?
If you do that the code should be a lot cleaner (you may need to work with mypy a bit though).

Reply:
As when in relying on triggering a `NameError` is quite hackish and reduces the readability of the code by quite a bit.